src/App.js

```diff
@@ -74,22 +75,45 @@ export default function App() {
             filterByGenres={filterByGenres}
             movies={fetchMovies}
             genres={genres}
-            currentUser={userSelection}
-            onYesWatch={handleYesWatch}
+            currentUser={currentUser}
+            onAddToWatchlist={handleAddToWatchlist}
+            onRemoveFromWatchlist={handleRemoveFromWatchlist}
           />
         </Route>
+        <Route path="/results">
+          <ResultsPage players={players} />
+        </Route>
       </Switch>
     </div>
   )
 
-  function handleYesWatch(movie, currentUser) {
+  function handleAddToWatchlist(movie, currentUser) {
     const player = players.find(player => player.name === currentUser.name)
     const index = players.indexOf(player)
-    setPlayers([
-      ...players.slice(0, index),
-      { ...player, movies: [...player.movies, movie] },
-      ...players.slice(index + 1),
-    ])
+    const isMovieInState = filterByGenres.filter(
+      item => item[movie.title] === movie.id
+    )
```

Comment:
You need to check if the movie is already in the list of movies of the player. Something like this I think?
```js
const isMovieInState = player.movies.filter(item => item.id === movie.id)
```